pytest-backend/exasol/pytest_backend/__init__.py

```diff
@@ -0,0 +1,218 @@
+                yield db_id
+            else:
+                # Create a temporary database and waite till it becomes operational
+                db = stack.enter_context(api_access.database(
+                    name=database_name,
+                    keep=keep,
+                    idle_time=timedelta(hours=idle_hours)))
+                api_access.wait_until_running(db.id)
+                yield db.id
+    else:
+        yield ''
+
+
+@pytest.fixture(scope="session")
+def backend_aware_onprem_database_params(use_onprem,
+                                         backend_aware_onprem_database,
+                                         exasol_config) -> dict[str, Any]:
+    if use_onprem:
+        return {
+            'dsn': f'{exasol_config.host}:{exasol_config.port}',
+            'user': exasol_config.username,
+            'password': exasol_config.password,
+            'websocket_sslopt': {'cert_reqs': ssl.CERT_NONE}
+        }
+    return {}
+
+
+@pytest.fixture(scope="session")
+def backend_aware_saas_database_params(use_saas,
+                                       saas_host,
+                                       saas_pat,
+                                       saas_account_id,
+                                       backend_aware_saas_database_id) -> dict[str, Any]:
+    if use_saas:
+        conn_params = get_connection_params(host=saas_host,
+                                            account_id=saas_account_id,
+                                            database_id=backend_aware_saas_database_id,
+                                            pat=saas_pat)
+        conn_params['websocket_sslopt'] = {'cert_reqs': ssl.CERT_NONE}
+        return conn_params
+    return {}
+
+
+@pytest.fixture(scope="session")
+def backend_aware_onprem_bucketfs_params(use_onprem,
+                                         backend_aware_onprem_database,
+                                         bucketfs_config) -> dict[str, Any]:
+    if use_onprem:
+        return {
+            'backend': _BACKEND_ONPREM,
+            'url': bucketfs_config.url,
+            'username': bucketfs_config.username,
+            'password': bucketfs_config.password,
+            'service_name': 'bfsdefault',
+            'bucket_name': 'default',
+            'verify': False
+        }
+    return {}
+
+
+@pytest.fixture(scope="session")
+def backend_aware_saas_bucketfs_params(use_saas,
+                                       saas_host,
+                                       saas_pat,
+                                       saas_account_id,
+                                       backend_aware_saas_database_id) -> dict[str, Any]:
+    if use_saas:
+        return {
+            'backend': _BACKEND_SAAS,
+            'url': saas_host,
+            'account_id': saas_account_id,
+            'database_id': backend_aware_saas_database_id,
+            'pat': saas_pat
+        }
+    return {}
+
+
+@pytest.fixture
```

Comment:
Should this fixture be scope "session", too?

Reply:
No, the `backend` is the per-test fixture. I cannot (or at least don't know how to) make it the session level.